WordPress/Classes/ViewRelated/Gutenberg/Utils/GutenbergFilesAppMediaSource.swift

```diff
@@ -14,7 +14,7 @@ class GutenbergFilesAppMediaSource: NSObject {
     func presentPicker(origin: UIViewController, filters: [Gutenberg.MediaType], allowedTypesOnBlog: [String], multipleSelection: Bool, callback: @escaping MediaPickerDidPickMediaCallback) {
         mediaPickerCallback = callback
         let documentTypes = GutenbergFilesAppMediaSource.getDocumentTypes(filters: filters, allowedTypesOnBlog: allowedTypesOnBlog)
-        let docPicker = UIDocumentPickerViewController(documentTypes: documentTypes, in: .import)
+        let docPicker = UIDocumentPickerViewController(forOpeningContentTypes: documentTypes.compactMap(UTType.init), asCopy: true)
```

Comment:
[Q] Is there an advantage to using `initForOpeningContentTypes:asCopy:` over `initForOpeningContentTypes:`? Do we need to copy the document?

Reply:
It's a good question. I used `asCopy` in the existing code in Site Media because it seems like a safer choice.

Documentation:

>  @param asCopy if true, the picker will give you access to a local copy of the document, otherwise you will have access to the original document

If you run an experiment, it does appear to place documents in different folders:

- `asCopy: true`:  file:///private/var/mobile/Containers/Data/Application/FCFA9E63-ECFC-4DB2-85EC-44F2EFF93879/tmp/com.automattic.jetpack-Inbox/file.pdf
- `asCopy: false`:/private/var/mobile/Library/Mobile%20Documents/com~apple~CloudDocs/Temp/file.pdf

If I use `asCopy: false` and test it on a physical device, I can't access the file contents:

```
The file “file.png” couldn’t be opened because you don’t have permission to view it.
```

I think you need to do some extra code to handle the permissions in case of the original URLs, something like [startAccessingSecurityScopedResource](https://developer.apple.com/documentation/foundation/url/1779698-startaccessingsecurityscopedreso), which I haven't tried. Making a copy sounds like a safer option.

Reply:
I also confirmed that the copy is deleted after you re-launch the app, so we are no wasting the resources.